tests/entry.sh

```diff
@@ -47,28 +56,50 @@ mount_dev() {
 	mount -t debugfs nodev /sys/kernel/debug
 }
 
-init() {
-	# trap the stop signal then send SIGTERM to user processes
-	trap signal_handler SIGRTMIN+3 SIGTERM
-
-	# echo error message, when executable file doesn't exist.
-	if CMD=$(command -v "$1" 2>/dev/null); then
-		shift
-		"$CMD" "$@" &
-		pid=$!
-		wait "$pid"
-		exit_code=$?
-		fg &> /dev/null || exit "$exit_code"
-	else
-		echo "Command not found: $1"
-		exit 1
-	fi
+init_systemd() {
+	GREEN='\033[0;32m'
+	echo -e "${GREEN}Systemd init system enabled."
+	for var in $(compgen -e); do
+		printf '%q=%q\n' "$var" "${!var}"
+	done > /etc/docker.env
+	echo 'source /etc/docker.env' >> ~/.bashrc
+
+	printf '#!/usr/bin/env bash\n exec ' > /etc/balenaApp.sh
+	printf '%q ' "$@" >> /etc/balenaApp.sh
+	chmod +x /etc/balenaApp.sh
+
+	mkdir -p /etc/systemd/system/balena.service.d
+	cat <<-EOF > /etc/systemd/system/balena.service.d/override.conf
+		[Service]
+		WorkingDirectory=$(pwd)
+	EOF
+
+	sleep infinity &
```

Comment:
do not comprehend all of this so cannot comment on whether it works or has side effects etc

Either way a 'sleep infinity' sounds like a strange thing.

Reply:
It keeps the initsystem running so that the docker container does not close right away when you run it.